### DISCUSSION TITLE
elementary.io broken purchase & download links

### DISCUSSION THREAD

**Discussion by chheller** [elementary/triage#391] [Q&A]
When I visit the main page, if I try to select any value other than $20, the value I select is ignored when the stripe page is loaded. After having paid $20, the now visible "Download" button still posts the same request to the `/api/create-checkout-session` endpoint, instead of linking to any download endpoint. The `api/create-checkout-session` endpoint doesn't appear to be handling the session token in the URL nor cookies, as it simply links back to the Strip checkout page again. If I modify the hidden value input in the form, then I can appropriately set the value to what I want. I've tested this is Firefox, Chrome, and Edge and all have the same behavior.

![image](https://github.com/elementary/triage/assets/2985885/ee50b366-e8cd-482e-b32d-0ad906273c0d)
![image](https://github.com/elementary/triage/assets/2985885/09b8aa83-18bf-4787-96dd-75b5d40da1b6)
![image](https://github.com/elementary/triage/assets/2985885/c8a1bd40-79fe-4041-9eb9-6c0cfdd09888)

**Comment by chheller** [comment]
Did a bit more digging and found that the website requires JQuery to run, and the JSDelivr CDN endpoint for JQuery was being blocked by the pihole on my network (though I've never run into an issue on another site). Leaving this for posterity for anyone else that might run into this edge-case, though it may be worth considering deploying a patch that integrates the payment and download forms without relying on JQuery since the form values seem like they could be implemented without JQuery or Javascript

**Reply by RMcNeely** [reply]
If you visit the main page without JavaScript there is a noscript element that runs if there's NO JavaScript allowed but I don't believe that shows up if you are just blocking a specific request.
